02.calendar/cal.js

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env node
+
+import minimist from "minimist";
+import _ from "lodash-es";
+
+const runCal = () => {
+  const baseDate = baseDateToMakeCalendar();
+  if (!baseDate.year || !baseDate.month) {
+    console.error(
+      "不正な入力です。年は半角自然数・月は半角1~12で入力してください。",
+    );
+    return;
+  }
+
+  const weeks = buildWeeks(baseDate.year, baseDate.month);
+  console.log(formatCalendar(baseDate.year, baseDate.month, weeks));
+};
+
+const baseDateToMakeCalendar = () => {
+  const { y: inputYear, m: inputMonth } = minimist(process.argv.slice(2));
+  const today = new Date();
+
+  return { year: year(inputYear, today), month: month(inputMonth, today) };
+};
+
+const year = (inputYear, today) => {
+  if (Number.isSafeInteger(inputYear)) return inputYear;
+  if (typeof inputYear === "undefined") return today.getFullYear();
+  return undefined;
+};
+
+const month = (inputMonth, today) => {
+  if (Number.isSafeInteger(inputMonth) && 1 <= inputMonth && inputMonth <= 12)
+    return inputMonth - 1;
+  if (typeof inputMonth === "undefined") return today.getMonth();
+  return undefined;
+};
+
+const buildWeeks = (year, month) => {
+  const firstDate = new Date(year, month, 1);
+  const lastDate = new Date(year, month + 1, 0);
+
+  const days = _.range(1, lastDate.getDate() + 1).map((day) => String(day));
+  days.unshift(...Array(firstDate.getDay()).fill(""));
```

Comment:
コメントし忘れていましたが、`unshift` を使うよりもスプレッド構文を使って新たな配列を生成するほうが、配列の先頭に空白を要素として追加したいという意図が明確になって読みやすくなると思います。

```suggestion
  [...Array(firstDate.getDay()).fill(""), ...days];
```

Reply:
ありがとうございます。
スプレッド構文で配列結合する方法を知りました。リファレンスのスプレッド構文のページでも、より良い連結方法として紹介されていたので、こちらを使用していこうと思います。